### ISSUE TITLE
Question - best configuration to schedule a repeating weekly, daily, or monthly alert from date and time.

### ISSUE THREAD

**Issue by bayhall** [katzer/cordova-plugin-local-notifications#1953]
I'm migrating this over from an older version of this plugin for a quick fix for an aging app.

I'm struggling to find the best configuration for a repeating daily, weekly or monthly notification based on the user selecting a start date and time.

The daily one appears to work, but what is the best format for weekly and monthly?

**This has worked for a daily alert:**

        cordova.plugins.notification.local.schedule({ 
            id: 16,
            title: 'My first daily scheduled notification 16',
            text: 'Thats pretty easy...',
            trigger: { firstAt: p_dtDateTime, every: { hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes()} },
            foreground: true
        });

**I think this should be correct for a weekly alert:**

        //weekly?
        cordova.plugins.notification.local.schedule({
            id: 17,
            title: 'My first weekly scheduled notification 17',
            text: 'Thats pretty easy...',
            trigger: { firstAt: p_dtDateTime, every: { weekday: p_dtDateTime.getDay(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },
            foreground: true
        });

**I'm struggling to find the most reliable format for a working monthly alert**

        //monthly?
        cordova.plugins.notification.local.schedule({
            id: 18,
            title: 'My first monthly scheduled notification 18',
            text: 'Thats pretty easy...',
            trigger: { firstAt: p_dtDateTime, every: 'month' },
            foreground: true
        });

**Comment by bayhall** [comment]
Currently these appear to be working except for Weekly (I can see they've at least been registered for the first alert and fire a local notification, and have persisted as a registered ID after firing). I'm selecting todays date, and a short time in the future:

**Daily - works**
 trigger: { firstAt: p_dtDateTime, every: { hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

**Weekly - doesn't work**
 trigger: { firstAt: p_dtDateTime, every: { weekday: p_dtDateTime.getDay(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

**Monthly - works**
trigger: { firstAt: p_dtDateTime, every: { day: p_dtDateTime.getDate(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

**Yearly - works**
   trigger: { firstAt: p_dtDateTime, every: { month: p_dtDateTime.getMonth() + 1, day: p_dtDateTime.getDate(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

**Comment by bayhall** [comment]
I've been able to get Weekly working by changing case of weekday to weekDay (after seeing other values were cased like weekOfMonth). 

To also have this work on Android and iOS I also needed to make sure the id for each was a string value.

I also noticed getIds crashed the app on iOS, but using getScheduledIds works (I use this to match system notifications with app stored db notifications, and remove any which are out of sync, ie not in the DB but are on the device).

Daily - works
trigger: { firstAt: p_dtDateTime, every: { hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

Weekly - works
trigger: { firstAt: p_dtDateTime, every: { **weekDay**: p_dtDateTime.getDay(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

Monthly - works
trigger: { firstAt: p_dtDateTime, every: { day: p_dtDateTime.getDate(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

Yearly - works
trigger: { firstAt: p_dtDateTime, every: { month: p_dtDateTime.getMonth() + 1, day: p_dtDateTime.getDate(), hour: p_dtDateTime.getHours(), minute: p_dtDateTime.getMinutes() } },

**Comment by bayhall** [comment]
Documentation is clear on scheduling local notifications - using a date and time I was able to set a schedule for a daily, weekly, monthly, and yearly alert.

**Comment by bhandaribhuminpfizer** [comment]
@bayhall  Thanks for your sample example of daily, weekly, monthly, and yearly alerts. I'm require another parameter to add for end notification till end date I also tried before but notification not works. 

My goal is to set the notification for 5 years @ 1 reminder per week.
`const trigger = { firstAt: triggerTime,before: endDate, unit: ELocalNotificationTriggerUnit.WEEK,every: { weekDay:event.start.getDay(), hour:  event.start.getHours(), minute:  event.start.getMinutes()}};`

**Comment by bhandaribhumin** [comment]
@bayhall so it will repeat continuously on weekday and how it is going to stop then?

**Comment by bhandaribhuminpfizer** [comment]
@bayhall  Thank you for your reply
Yes, you are right. I already set delete by ID functionality in my app.
But if I want to stop the reminder after 1 or 2 years without user interaction how it is possible?

My goal is to set the notification for `5 years @ 1 reminder per week`

Example 

Goto: https://jakubroztocil.github.io/rrule/

new RRule({
  freq: RRule.WEEKLY,
  dtstart: new Date(Date.UTC(2021, 7, 13, 13, 44, 0)),
  until: new Date(Date.UTC(2025, 7, 13, 13, 44, 0)),
  count: 265,
  interval: 1,
  wkst: RRule.WE
});

I tried using RRule object which is 265 count but iOS only allow a schedule notification limit is 64 so try using weekday but not able to stop notification after 5 years.
 
```

 let trigger:any = {
          firstAt: triggerTime,
          every:
          {
            weekday: event.start.getDay(),
            hour: event.start.getHours(),
            minute: event.start.getMinutes()
          }
        };
        if(this.platform.is('android')){
          trigger.before = new Date(moment(event.start).add(5, 'years').format());
        }
```

**Comment by bhandaribhumin** [comment]
@bayhall   Thank you for your reply
Yes, you are right. I already set delete by ID functionality in my app.
But if I want to stop the reminder after 1 or 2 years without user interaction how it is possible?

My goal is to set the notification for `5 years @ 1 reminder per week`

Example 

Goto: https://jakubroztocil.github.io/rrule/

new RRule({
  freq: RRule.WEEKLY,
  dtstart: new Date(Date.UTC(2021, 7, 13, 13, 44, 0)),
  until: new Date(Date.UTC(2025, 7, 13, 13, 44, 0)),
  count: 265,
  interval: 1,
  wkst: RRule.WE
});

I tried using RRule object which is 265 count but iOS only allows a schedule notification limit is 64 so try using weekday but not able to stop notification after 5 years.
 
```

 let triggersObj:any = {
          firstAt: triggerTime,
          every:
          {
            weekday: day.getDay(),
            hour: day.getHours(),
            minute: day.getMinutes()
          }
        };
        if(this.platform.is('android')){
          triggersObj.before = new Date(moment(day).add(5, 'years').format());
        }
```
